src/Features/Core/Portable/Diagnostics/DiagnosticResultSerializer.cs

```diff
@@ -140,7 +143,16 @@ internal static class DiagnosticResultSerializer
             for (var i = 0; i < count; i++)
             {
                 var documentId = DocumentId.ReadFrom(reader);
-                var diagnostics = serializer.ReadFrom(reader, project.GetDocument(documentId), cancellationToken);
+                var document = project.GetDocument(documentId);
+
+                var diagnostics = serializer.ReadFrom(reader, document, cancellationToken);
+
+                if (document?.SupportDiagnostics() == false)
```

Comment:
what is an example of a document that doesn't support diagnsotics?  Also, checking for null here in this way seems weird.  It seems that if we got a null document, we sholudn't even bother reading diagnostics...

Reply:
document can be null if documentId no longer exist in the solution. that's existing behavior. I will keep existing behavior.

whether a document supporting diagnostics or not is document service provider's decision. we just follow whatever decision they choose. for razor, they choose supporting diagnostics for open files but not closed files. later it might change but again, it's thier decision.

Reply:
please, don't argue on design decision we made. if you want to argue on decision, make an issue and argue in that issue not this PR.

Reply:
Heejae, designs are a place where things can be initially thought out, but are often lacking information due to being so separated from code.  When a PR comes out it's not intended to be a case where tehre is blind adherence to previous decisions.  it's a place where the design decisions can now be assessed against the cold reality of the codebase.  If something is strange/faulty, we should discuss it and assess *with the most context* to understand appropriately.

If you want to open another issue, that's fine with me.  however, this PR would then be gated on that issue reaching a resolved state.  It would be irresponsible to not do that as there is nothing about prior-to-code design-decision that outweighs problems found in a PR.

Reply:
design question we are talking about here is this

can diagnostics be disabled for a document. we decided razor should be able to control that. 

this is alraedy decided on meeting. not going to change that in this PR. if we want to start talk, start that in different issue.

anyway, I will now skip those kind of question.

Reply:
>  we sholudn't even bother reading diagnostics...

we need to read/remove data from stream, otherwise, data in stream will be in corrupted state.